src/libfuncs/bool.rs

```diff
@@ -1,3 +1,316 @@
+        DenseI64ArrayAttribute::new(context, &[0]),
+        tag_ty,
+        location,
+    ));
+    let rhs_tag = op.result(0)?.into();
+
+    let op = match binop {
+        BoolOp::And => entry.append_operation(arith::andi(lhs_tag, rhs_tag, location)),
+        BoolOp::Xor => entry.append_operation(arith::xori(lhs_tag, rhs_tag, location)),
+        BoolOp::Or => entry.append_operation(arith::ori(lhs_tag, rhs_tag, location)),
+    };
+    let new_tag_value = op.result(0)?.into();
+
+    let op = entry.append_operation(llvm::insert_value(
+        context,
+        lhs,
+        DenseI64ArrayAttribute::new(context, &[0]),
+        new_tag_value,
+        location,
+    ));
+    let value = op.result(0)?.into();
+
+    entry.append_operation(helper.br(0, &[value], location));
+    Ok(())
+}
+
+/// Generate MLIR operations for the `bool_not_impl` libfunc.
+pub fn build_bool_not<'ctx, 'this, TType, TLibfunc>(
+    context: &'ctx Context,
+    registry: &ProgramRegistry<TType, TLibfunc>,
+    entry: &'this Block<'ctx>,
+    location: Location<'ctx>,
+    helper: &LibfuncHelper<'ctx, 'this>,
+    _metadata: &mut MetadataStorage,
+    info: &SignatureOnlyConcreteLibfunc,
+) -> Result<()>
+where
+    TType: GenericType,
+    TLibfunc: GenericLibfunc,
+    <TType as GenericType>::Concrete: TypeBuilder<TType, TLibfunc, Error = CoreTypeBuilderError>,
+    <TLibfunc as GenericLibfunc>::Concrete: LibfuncBuilder<TType, TLibfunc, Error = Error>,
+{
+    let enum_ty = registry.get_type(&info.param_signatures()[0].ty)?;
+    let tag_bits = enum_ty
+        .variants()
+        .unwrap()
+        .len()
+        .next_power_of_two()
+        .trailing_zeros();
+    let tag_ty = IntegerType::new(context, tag_bits).into();
+
+    let value = entry.argument(0)?.into();
+
+    let op = entry.append_operation(llvm::extract_value(
+        context,
+        value,
+        DenseI64ArrayAttribute::new(context, &[0]),
+        tag_ty,
+        location,
+    ));
+    let tag_value = op.result(0)?.into();
+
+    let op = entry.append_operation(arith::constant(
+        context,
+        Attribute::parse(context, &format!("1 : {tag_ty}")).unwrap(),
+        location,
+    ));
+    let const_1 = op.result(0)?.into();
+
+    let op = entry.append_operation(arith::xori(tag_value, const_1, location));
+    let new_tag_value = op.result(0)?.into();
+
+    let op = entry.append_operation(llvm::insert_value(
+        context,
+        value,
+        DenseI64ArrayAttribute::new(context, &[0]),
+        new_tag_value,
+        location,
+    ));
+    let value = op.result(0)?.into();
+
+    entry.append_operation(helper.br(0, &[value], location));
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
+    use crate::utils::test::{load_cairo, run_program};
+    use serde_json::json;
+
+    #[test]
+    fn run_not() {
+        let program = load_cairo!(
+            use array::ArrayTrait;
+
+            fn run_test() -> bool {
+                true == true
+            }
+        );
```

Comment:
Calling `load_cairo!` is extremely expensive (that was the whole point in changing the way testing works). Maybe you could use parameters instead of constants and keep one `load_cairo!` invocation per test? Or even better, one invocation per multiple tests.